From 117463102379bfb419e4f9ac1975cbbcd6bbd7a6 Mon Sep 17 00:00:00 2001
From: Stephan Bergmann <sbergman@redhat.com>
Date: Mon, 25 Nov 2013 18:01:22 +0100
Subject: [PATCH] rhbz#1030009 Remove global XModel ref

...that is apparently effectively unused since
2627bdaddc6f694dc7fc6ba347a305af0f88ee2c "INTEGRATION: CWS fwk02ea: #i28639#
always used supplied status indicator" anyway but causes crashes during exit
when it holds an SwXTextDocument ref.

Change-Id: I542545c793855b41731de72650bf07d5748fcb9f
(cherry picked from commit a7464edb0a4368db02b1298b5710cf98b17e282b)
---
 filter/source/xmlfilteradaptor/XmlFilterAdaptor.cxx | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/filter/source/xmlfilteradaptor/XmlFilterAdaptor.cxx b/filter/source/xmlfilteradaptor/XmlFilterAdaptor.cxx
index c0b017c..a486cdf 100644
--- a/filter/source/xmlfilteradaptor/XmlFilterAdaptor.cxx
+++ b/filter/source/xmlfilteradaptor/XmlFilterAdaptor.cxx
@@ -36,7 +36,6 @@
 #include <com/sun/star/frame/XConfigManager.hpp>
 #include <com/sun/star/xml/XImportFilter.hpp>
 #include <com/sun/star/xml/XExportFilter.hpp>
-#include <com/sun/star/frame/XModel.hpp>
 #include <com/sun/star/frame/XController.hpp>
 #include <com/sun/star/task/XStatusIndicator.hpp>
 #include <com/sun/star/task/XStatusIndicatorFactory.hpp>
@@ -66,8 +65,6 @@ using namespace ::com::sun::star::task;
 
 #define MAP_LEN(x) x, sizeof(x) - 1
 
-Reference< com::sun::star::frame::XModel > xModel;
-
 sal_Bool SAL_CALL XmlFilterAdaptor::importImpl( const Sequence< ::com::sun::star::beans::PropertyValue >& aDescriptor )
     throw (RuntimeException)
 {
@@ -341,9 +338,6 @@ void SAL_CALL XmlFilterAdaptor::setSourceDocument( const Reference< ::com::sun::
 {
     meType = FILTER_EXPORT;
     mxDoc = xDoc;
-        com::sun::star::uno::Reference< com::sun::star::frame::XModel >rModel ( com::sun::star::uno::Reference< com::sun::star::frame::XModel >::query( xDoc ) );
-    xModel=rModel;
-
 }
 
 // XImporter
@@ -352,7 +346,6 @@ void SAL_CALL XmlFilterAdaptor::setTargetDocument( const Reference< ::com::sun::
 {
     meType = FILTER_IMPORT;
     mxDoc = xDoc;
-    //xModel = uno::Reference< frame::XModel >::query( xDoc );
 }
 // XInitialization
 void SAL_CALL XmlFilterAdaptor::initialize( const Sequence< Any >& aArguments )
-- 
1.8.3.1

